### ISSUE TITLE
Layouts in Tabs behaving unexpectedly with relative sizing modes

### ISSUE THREAD

**Issue by csbrace** [bokeh/bokeh#8043]
Hi all,
I've recently run into an issue while trying to implement multi-level tabs in Bokeh, and after some research I think this may be a bug in how tabs handle relative sizing, and it may be one that's [been around for a few months.](https://groups.google.com/a/continuum.io/forum/#!msg/bokeh/Py1VpXihtmA/1murm_n_BgAJ)

Specifically, when you put figures in tabs, then put those tabs in a gridplot/row/column, then put that layout object in another tab, showing the top level tab will give you different buggy looking behaviour depending on the relative sizing mode specified.

Attached is (I hope) minimal sample code to reproduce

```
from bokeh.io import output_file, show
from bokeh.layouts import row, column, gridplot
from bokeh.plotting import figure
from bokeh.models.widgets import Panel, Tabs

output_file("tabIssue.html")

# random setup, ignore
x = list(range(11))
y0 = x
y1 = [10 - i for i in x]
y2 = [abs(i - 5) for i in x]
y3 = [abs(i + 5) for i in x]

# create two plots for first tab
f1 = figure(title="X Plant to Z Plant", x_axis_label = 'Created Date', 
            y_axis_label = "Total Items", plot_width = 650, plot_height = 650)
f1.circle(x, y0) # using non-fixed plot_widths eg scale_width in place of fixed 
# sizes makes the plots disappear entirely
f2 = figure(title="X Plant to Z Plant", x_axis_label = 'Created Date', 
            y_axis_label = "Total Weight", plot_width = 650, plot_height = 650)
f2.circle(x, y1)

# create two plots for second tab
f3 = figure(title="Y Plant to Z Plant", x_axis_label = 'Created Date', 
            y_axis_label = "Total Items", plot_width = 650, plot_height = 650)
f3.circle(x, y2)
f4 = figure(title="Y Plant to Z Plant", x_axis_label = 'Created Date', 
            y_axis_label = "Total Items", plot_width = 650, plot_height = 650)
f4.circle(x, y3)

p1 = Panel(child=f1, title="Plot 1", sizing_mode='scale_width')
p2 = Panel(child=f2, title="Plot 2", sizing_mode='scale_width')

p3 = Panel(child=f3, title="Plot 3", sizing_mode='scale_width')
p4 = Panel(child=f4, title="Plot 4", sizing_mode='scale_width')

t1 = Tabs(tabs = [p1, p2], sizing_mode='scale_width')
t2 = Tabs(tabs = [p3, p4], sizing_mode='scale_width')

g1 = gridplot([t1,t2], ncols=2, merge_tools=False) 
""" Adding sizing_mode="scale_width" breaks it even worse, puts the figures right 
on top of each other and makes the inner tabs dissapear. All the same issues persist
with both rows and columns, so I don't think this is gridplot specific"""

gPanel = Panel(child=g1, title="Gridplot", sizing_mode='scale_width')

t3 = Tabs(tabs = [gPanel], sizing_mode='scale_width')

show(t3)
```
This example is the least broken I can make this look. As noted in the code, replacing the fixed scaling on the figures with any relative makes them disappear (can demonstrate by changing the last line to t1 or t2 from t3, and adding ", sizing_mode='scale_width'" to the figures), and adding it to the gridplot makes the figures display on top of each other.

I'm no expert with Bokeh or even Python, and this is my first issue here, so please let me know if there's any other info I can provide, or if I'm missing something obvious. I just want to have multi-level tabs working for my plots, so I can display my company's shipping data in a sensible way.

TL;DR: I don't think Tabs are playing nice with the relative layouts of their children.

I'm using Python 3.6.4 with Bokeh 0.12.6, displaying on Chrome 67.0.3396.87

**Comment by bryevdv** [comment]
There are lots of open issues with layout, and with tabs specifically. This is almost certainly a duplicate of an existing issue, but I will have to try and track it down. If you need something more than a veyr basic layout (and multi-level tabs sounds like that). I would suggest for now using the various functions in `bokeh.embed` to embed individual plots in your own HTML template (and handle the tab panes in the template)

**Comment by csbrace** [comment]
Fair enough, that. I feel like two layers of tabs is a fairly common use case that would be nice to handle internally, but for now I think I'll cheat a little bit and just use absolute sizing a la
'''
t1.width = 700
t2.width = 700
'''
to make my prototype work visually, and I'll embed everything like you suggest for the final product.
Also, looking through other examples from my own code, I think I may have over-complicated the problem: it looks like tabs may just not be communicating their sizing down to their non-plot children at all, and I just happened to only run into this as an issue when creating my multi-layer tabs. Then again, I'm wildly speculating: my HTML/JS knowledge is very minimal.
Thanks for the clarification!

**Comment by bryevdv** [comment]
My pleasure, I am going to go ahead and close this discussion as I do believe it is covered by other layout/tabs issues already. Resources are limited but I hope we are able to turn to some of these soon.

**Comment by brcnblc** [comment]
I experience a similar issue. When I render column layouts with property sizing_mode = 'scale_width' , I encounter empty tab pages. Same thing happens if sizing mode is 'scale_height' or 'stretch_both'.

Page is rendered normally if I use 'fixed' as sizing_mode.

Bokeh version is. 1.0.3

Column1 includes 1 external toolbar and 2 figures, Column 2 includes 1 external toolbar and 1 figure.
Each column is rendered properly with relative sizing modes if they are shown separately with show. The problem persists even I drop the external toolbar from the column array. 

The only object that can be seen is the half of the bokeh logo at left side of the screen.

Best Regards.
